kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaHistoryManager.java

```diff
@@ -40,44 +41,66 @@ private DeltaHistoryManager() {}
   private static final Logger logger = LoggerFactory.getLogger(DeltaHistoryManager.class);
 
   /**
-   * Returns the latest recreatable commit that happened at or before {@code timestamp}. If the
-   * provided timestamp is after the timestamp of the latest version of the table throws an
-   * exception. If the provided timestamp is before the timestamp of the earliest version of the
-   * table throws an exception.
+   * Returns the latest commit that happened at or before {@code timestamp}.
+   *
+   * If the timestamp is outside the range of [earliestCommit, latestCommit] then use
+   * parameters {@code canReturnLastCommit} and {@code canReturnEarliestCommit} to control whether
+   * an exception is thrown or the corresponding earliest/latest commit is returned.
    *
    * @param engine instance of {@link Engine} to use
    * @param logPath the _delta_log path of the table
    * @param timestamp the timestamp find the version for in milliseconds since the unix epoch
-   * @return the active recreatable commit version at the provided timestamp
+   * @param mustBeRecreatable whether the state at the returned commit should be recreatable
+   * @param canReturnLastCommit whether we can return the latest version of the table if the
+   *                            provided timestamp is after the latest commit
+   * @param canReturnEarliestCommit whether we can return the earliest version of the table if the
+   *                                provided timestamp is before the earliest commit
+   * @throws KernelException if the provided timestamp is before the earliest commit and
+   *                         canReturnEarliestCommit is false
+   * @throws KernelException if the provided timestamp is after the latest commit and
+   *                         canReturnLastCommit is false
    * @throws TableNotFoundException when there is no Delta table at the given path
    */
-  public static long getActiveCommitAtTimestamp(Engine engine, Path logPath, long timestamp)
-      throws TableNotFoundException {
+  public static Commit getActiveCommitAtTimestamp(
+      Engine engine,
+      Path logPath,
+      long timestamp,
+      boolean mustBeRecreatable,
```

Comment:
this is only ever invoked with `mustBeRecreatable =false`. do we ever expect it to be invoked with `mustBeRecreatable=true`? if not, can we remove this param and then cut down our testing space in half?

Reply:
We use it with `mustBeRecreatable=true` for snapshot time travel queries

Reply:
Ah, missed that! Thanks!